### PR TITLE
[GISel] Restrict G_BSWAP to multiples of 16 bits.

### PR DESCRIPTION
**Pull request by topperc**
This is consistent with the IR verifier and SelectionDAG's getNode.

Update tests accordingly. I tried to keep some coverage of non-pow2 when possible. X86 didn't like a G_UNMERGE_VALUES from s48 to 3 s16 that got created when I tried s48.

### CODE REVIEW

File: llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir [hunk 1]
```diff
@@ -4,35 +4,6 @@
 
 # test bswap for s16, s17, s32, and s64
 
-...
----
-name:            test_bswap17
```
Comment (michaelmaitland) [inline llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir:9]: change to bswap48 instead of removing?

Comment (michaelmaitland) [reply llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir:9]: Didn't see your PR description. NVM

Comment (topperc) [reply llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir:9]: As I wrote in my description, I tried. I hit some other legalizer issue.

**Review by michaelmaitland — APPROVED** [review]
LGTM

File: llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir (hunk 1 above)
Comment (phoebewang) [reply llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir:9]: cc @e-kud

**Review by arsenm — COMMENTED** [review]
Should get tested in test/MachineVerifier

File: llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir (hunk 1 above)
Comment (e-kud) [reply llvm/test/CodeGen/X86/GlobalISel/legalize-bswap.mir:9]: Yeah, `X86` implementation was copied from `AArch64` that also fails on the following mir with the same assert
```
 %0:_(s48) = IMPLICIT_DEF
 %1:_(s16), %2:_(s16), %3:_(s16) = G_UNMERGE_VALUES %0:_(s48)
 %4:_(s32) = G_ANYEXT %1(s16)
 $w0 = COPY %4(s32)
 RET_ReallyLR implicit $w0
```

**Pull request merged**